src/train.py

```diff
@@ -0,0 +1,26 @@
+'''
+Contains the code for training the encoder/decoders, including:
+    - Cross model loss
+    - Supervised loss
+    - Denoising loss
+    - Discriminator loss
+'''
+
+def train():
```

Comment:
This is great. Try to merge this in ASAP. I will try to update the dataset processing code.